tests/framework/Optimizers/GradientDescent/constrain_boundary.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>framework/Optimizers.ConstrainBoundary</name>
+    <author>talbpaul</author>
+    <created>2020-01-30</created>
+    <classesTested>Optimizer</classesTested>
+    <description>
+      This test uses a multidimensional linear function with optima outside the domain space to test
+      boundary constraint handling.
```

Comment:
it should be a requirement tests

Reply:
Added R-RM-2 for constraining the space of variables used to find min/max goal functions.